Makefile.archlinux

```diff
@@ -143,23 +151,53 @@ dist-copy-out:
 
 ### Additional targets
 
+# Sign packages
+sign: sign_client = $(if $(GNUPG),$(GNUPG),gpg)
+sign:
+	@if [ -d $(ORIG_SRC)/$(OUTPUT_DIR) ]; then \
+		cd $(ORIG_SRC)/$(OUTPUT_DIR); \
+		for filename in *.pkg.tar.zst; do\
+			echo $$filename; \
+			$(sign_client) --yes --local-user "$(SIGN_ARCHLINUX_KEY)" --detach-sign -o "$$filename.sig" "$$filename";\
+			ln -f -t $(BUILDER_REPO_DIR)/pkgs "$$filename.sig";\
+		done; \
+	fi
+
+
 # Copies requested packages (based on PACKAGE_SET, COMPONENT, DIST) to
 # requested repository (UPDATE_REPO)
 update-repo:
 ifndef UPDATE_REPO
 	$(error "You need to specify destination repo in UPDATE_REPO variable")
 endif
-	mkdir -p $(UPDATE_REPO)/pkgs
-	for arch_build_dir in $(ARCH_BUILD_DIRS); do\
-	  pkgnames=`cat $(ORIG_SRC)/$$arch_build_dir/PKGBUILD | grep pkgname | cut -d "=" -f 2 | tr -d '()"'`;\
-	  for pkgname in $$pkgnames; do\
-	    ln -f $(ORIG_SRC)/pkgs/$$pkgname-*.pkg.tar.* $(UPDATE_REPO)/pkgs/;\
-	  done;\
-	done;\
-
-# Sign packages
-sign:
-	cd $(ORIG_SRC)
-	for filename in pkgs/*.pkg.tar.* ; do
-	  qubes-gpg-client-wrapper --detatch-sign "$filename" > "$filename.sig"
+ifeq (,$(PACKAGE_LIST))
+	@true
+else
+ifdef SNAPSHOT_FILE
+	@echo -n > $(SNAPSHOT_FILE)
+endif
+	mkdir -p $(UPDATE_REPO)/pkgs; \
+	for package in $(PACKAGE_LIST); do\
+		cd $(ORIG_SRC); \
```

Comment:
This is risky, UPDATE_REPO may be a path relative to the current directory - changing it will break `ln` calls below. Better prepend ORIG_SRC in relevant places below.

Reply:
I've added ORIG_SRC in the `ln`. SNAPSHOT_FILE seems to be absolute path.